gateway/api/connections/connections.go

```diff
@@ -382,3 +391,532 @@ func FetchByName(ctx pgrest.Context, connectionName string) (*models.Connection,
+		return
+	}
+
+	userAgent := apiutils.NormalizeUserAgent(c.Request.Header.Values)
+
+	client, err := clientexec.New(&clientexec.Options{
+		OrgID:          ctx.GetOrgID(),
+		ConnectionName: conn.Name,
+		BearerToken:    getAccessToken(c),
+		UserAgent:      userAgent,
+		// it sets the execution to perform plain executions
+		Verb: proto.ClientVerbPlainExec,
+	})
+	if err != nil {
+		log.Error(err)
+		c.JSON(http.StatusInternalServerError, gin.H{"message": err.Error()})
+		return
+	}
+
+	respCh := make(chan *clientexec.Response)
+	go func() {
+		defer func() { close(respCh); client.Close() }()
+		select {
+		case respCh <- client.Run([]byte(script), nil):
+		default:
+		}
+	}()
+
+	timeoutCtx, cancelFn := context.WithTimeout(context.Background(), time.Second*50)
+	defer cancelFn()
+	select {
+	case outcome := <-respCh:
+		if outcome.ExitCode != 0 {
+			c.JSON(http.StatusInternalServerError, gin.H{"message": fmt.Sprintf("command failed: %s", outcome.Output)})
+			return
+		}
+
+		var databases []string
+		var err error
+
+		if pb.ToConnectionType(conn.Type, conn.SubType.String) == pb.ConnectionTypeMongoDB {
+			var result []map[string]interface{}
+			if err := json.Unmarshal([]byte(outcome.Output), &result); err != nil {
+				c.JSON(http.StatusInternalServerError, gin.H{"message": fmt.Sprintf("failed to parse MongoDB response: %v", err)})
+				return
+			}
+			for _, db := range result {
+				if dbName, ok := db["database_name"].(string); ok {
+					databases = append(databases, dbName)
+				}
+			}
+		} else {
+			databases, err = parseCommandOutput(outcome.Output)
+			if err != nil {
+				c.JSON(http.StatusInternalServerError, gin.H{"message": fmt.Sprintf("failed to parse response: %v", err)})
+				return
+			}
+		}
+
+		c.JSON(http.StatusOK, DatabaseListResponse{
+			Databases: databases,
+		})
+	case <-timeoutCtx.Done():
+		client.Close()
+		log.Infof("runexec timeout (50s), it will return async")
+		c.JSON(http.StatusRequestTimeout, gin.H{"message": "Request timed out"})
+	}
+}
+
+// SchemaResponse representa a estrutura completa do schema
+type SchemaResponse struct {
+	Schemas []Schema `json:"schemas"`
+}
+
+type Schema struct {
+	Name   string  `json:"name"`
+	Tables []Table `json:"tables"`
+	Views  []View  `json:"views"`
+}
+
+type Table struct {
+	Name    string   `json:"name"`
+	Columns []Column `json:"columns"`
+	Indexes []Index  `json:"indexes"`
+}
+
+type View struct {
+	Name    string   `json:"name"`
+	Columns []Column `json:"columns"`
+}
+
+type Column struct {
+	Name         string `json:"name"`
+	Type         string `json:"type"`
+	Nullable     bool   `json:"nullable"`
+	DefaultValue string `json:"default_value,omitempty"`
+	IsPrimaryKey bool   `json:"is_primary_key"`
+	IsForeignKey bool   `json:"is_foreign_key"`
+}
+
+type Index struct {
+	Name      string   `json:"name"`
+	Columns   []string `json:"columns"`
+	IsUnique  bool     `json:"is_unique"`
+	IsPrimary bool     `json:"is_primary"`
+}
```

Comment:
API types should be in the openapi package: `gateway/api/openapi/openapi.go`. Make sure to document the types properly and give proper examples to them.